setup.py

```diff
@@ -62,23 +81,29 @@ def write_version_py(filename='traitsui/_version.py'):
 if not is_released:
     version = full_version
 """
+    if filename is None:
+        # correctly generate relative path
+        base_dir = os.path.dirname(__file__)
+        filename = os.path.join(base_dir, 'traitsui', '_version.py')
+
     # Adding the git rev number needs to be done inside
     # write_version_py(), otherwise the import of traits._version messes
     # up the build under Python 3.
     fullversion = VERSION
     if os.path.exists('.git'):
         git_rev, dev_num = git_version()
-    elif os.path.exists('traitsui/_version.py'):
+    elif os.path.exists(filename):
         # must be a source distribution, use existing version file
         try:
-            from traitsui._version import git_revision as git_rev
-            from traitsui._version import full_version as full_v
+            data = read_module('_version')
+            git_rev = data['git_revision']
+            fullversion_source = data['full_version']
         except ImportError:
```

Comment:
`ImportError` is not longer likely here, correct?

Reply:
Yes - given that we are checking for the existence of the file, errors are most likely to be about unexpected content in the file because someone has messed with it (SyntaxErrors, KeyErrors, etc.) or something wrong with the filesystem (IOErrors and OSErrors of various sorts, I would guess).

Possibly best just to catch `Exception`, print a message, and re-raise?